flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCTableSourceSinkFactoryITCase.java

```diff
@@ -0,0 +1,298 @@
+		env.execute();
+
+		List<String> expected = new ArrayList<>();
+		for (TestEntry entry : TEST_DATA) {
+			expected.add(entry.id + "," + entry.title + "," + entry.author + "," + entry.price + "," + entry.qty);
+		}
+		StreamITCase.compareWithList(expected);
+		StreamITCase.clear();
+	}
+
+	@Test
+	public void testJDBCScanSourceWithProjection() throws Exception {
+		Map<String, String> properties = getBasicProperties(INPUT_TABLE);
+
+		final StreamTableSource<?> source = TableFactoryService.find(StreamTableSourceFactory.class, properties)
+			.createStreamTableSource(properties);
+
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+		tEnv.registerTableSource("books", source);
+
+		Table result = tEnv.sqlQuery("select id, title, price from books");
+		DataStream<Row> resultSet = tEnv.toAppendStream(result, Row.class);
+		resultSet.addSink(new StreamITCase.StringSink<>());
+		env.execute();
+
+		List<String> expected = new ArrayList<>();
+		for (TestEntry entry : TEST_DATA) {
+			expected.add(entry.id + "," + entry.title + "," + entry.price);
+		}
+		StreamITCase.compareWithList(expected);
+		StreamITCase.clear();
+	}
+
+	@Test
+	public void testJDBCScanSourceWithFieldMapping() throws Exception {
+		Map<String, String> properties = getBasicProperties(INPUT_TABLE);
+		properties.put(Schema.SCHEMA + ".0." + Schema.SCHEMA_NAME, "my_id");
+		properties.put(Schema.SCHEMA + ".0." + Schema.SCHEMA_FROM, "id");
+		properties.put(Schema.SCHEMA + ".2." + Schema.SCHEMA_NAME, "my_author");
+		properties.put(Schema.SCHEMA + ".2." + Schema.SCHEMA_FROM, "author");
+		properties.put(Schema.SCHEMA + ".4." + Schema.SCHEMA_NAME, "my_qty");
+		properties.put(Schema.SCHEMA + ".4." + Schema.SCHEMA_FROM, "qty");
+
+		final StreamTableSource<?> source = TableFactoryService.find(StreamTableSourceFactory.class, properties)
+			.createStreamTableSource(properties);
+
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+		tEnv.registerTableSource("books", source);
+
+		Table result = tEnv.sqlQuery("select my_id, title, my_author, price, my_qty from books");
+		DataStream<Row> resultSet = tEnv.toAppendStream(result, Row.class);
+		resultSet.addSink(new StreamITCase.StringSink<>());
+		env.execute();
+
+		List<String> expected = new ArrayList<>();
+		for (TestEntry entry : TEST_DATA) {
+			expected.add(entry.id + "," + entry.title + "," + entry.author + "," + entry.price + "," + entry.qty);
+		}
+		StreamITCase.compareWithList(expected);
+		StreamITCase.clear();
+	}
+
+	@Test
+	public void testJDBCLookupSource() throws Exception {
+		testJDBCLookupSourceImpl(false);
+	}
+
+	@Test
+	public void testJDBCLookupSourceWithCache() throws Exception {
+		testJDBCLookupSourceImpl(true);
+	}
+
+	private void testJDBCLookupSourceImpl(boolean useCache) throws Exception {
```

Comment:
Does `SCHEMA_FROM` can work with LookupSource?

Reply:
After some consideration I discover that `SCHEMA_FROM` api is sort of confusing. As only kafka source implements this api currently, I decide to not implement this temporarily.